resources/generate_spec_files.py

```diff
@@ -1,101 +1,182 @@
+    var_names.append(var_name)
+    var_contents[var_name] = json_str
 
-        # create header file
-        header_file = Path(f"./src/spec/{ns['name'].replace('-', '_')}.hpp").resolve()
-        with open(header_file, 'w') as fo:
-            fo.write('#pragma once\n\n')
-            fo.write('#include <array>\n#include <string>\n#include <string_view>\n\n')
-            fo.write(f'namespace AQNWB::SPEC::{ns["name"].upper().replace("-", "_")}\n{{\n\n')
-            fo.write(f'const std::string version = "{ns["version"]}";\n\n')
+def process_namespace_file(namespace_file: Path, output_dir: Path, chunk_size: int) -> None:
+    """
+    Process a namespace file and generate the corresponding C++ header files.
+    
+    Args:
+        namespace_file (Path): Path to the namespace file.
+        output_dir (Path): Directory to output the generated header files.
+        chunk_size (int): Size of the chunks for splitting large JSON strings.
+    """
+    logger.info(f"Processing namespace file: {namespace_file}")
+    try:
+        namespace = load_file(namespace_file)
+    except Exception as e:
+        logger.error(f"Failed to process namespace file {namespace_file}: {e}")
+        return
 
-        # load and convert schema files
+    for ns in namespace['namespaces']:
+        header_file = output_dir / f"{ns['name'].replace('-', '_')}.hpp"
         var_names = []
+        var_contents = {}
         for s in ns['schema']:
             if 'source' in s:
-                # load file
-                schema_file = file.parent / s['source']
-                with open(schema_file) as f:
-                    spec = yaml.load(f)
-
-                # convert to cpp string
-                print(f'Generating file {header_file} - {s["source"]}')
-                json_str = json.dumps(spec, separators=(',', ':'))
-                chunk_size = 16000  # Adjust the chunk size as needed
-                if len(json_str) > chunk_size:
-                    # Split string into chunks if needed
-                    chunks = [json_str[i:i + chunk_size] for i in range(0, len(json_str), chunk_size)]
-                    
-                    var_name = s['source'].replace('.yaml', '').replace('.', '_')
-                    chunk_var_names = []
-                    for i, chunk in enumerate(chunks):
-                        chunk_var_name = f'{var_name}_part{i}'
-                        with open(header_file, 'a') as fo:
-                            fo.write(f'constexpr std::string_view {chunk_var_name} = R"delimiter({chunk})delimiter";\n')
-                        chunk_var_names.append(chunk_var_name)
+                schema_file = namespace_file.parent / s['source']
+                # If the schem file is missing, check if it was converted to JSON/YAML
+                if not os.path.exists(schema_file):
+                    for ext in ['.yaml', '.yml', '.json']:
```

Comment:
```suggestion
                    for ext in ['.yaml', '.yml']:
```

See previous comment about supporting the .json files